src/optimizer.jl

```diff
@@ -0,0 +1,179 @@
+# Supported Functions
+const SF = Union{MOI.SingleVariable, 
+                 MOI.ScalarAffineFunction{Float64}, 
+                 MOI.VectorOfVariables, 
+                 MOI.VectorAffineFunction{Float64}}
+
+# Supported Sets
+const SS = Union{MOI.EqualTo{Float64}, MOI.GreaterThan{Float64}, MOI.LessThan{Float64}, 
+                 MOI.Zeros, MOI.Nonnegatives, MOI.Nonpositives, 
+                 MOI.SecondOrderCone, MOI.RotatedSecondOrderCone,
+                 MOI.ExponentialCone, MOI.DualExponentialCone,
+                 MOI.PowerCone, MOI.DualPowerCone,
+                 MOI.PositiveSemidefiniteConeTriangle}
+
+mutable struct DualOptimizer <: MOI.AbstractOptimizer
+    dual_problem::Union{Nothing, Dualization.DualProblem}
+    dual_optimizer::Union{Nothing, MOI.AbstractOptimizer}
+    dual_optimizer_idx_map::Union{Nothing, MOIU.IndexMap}
+
+    function DualOptimizer(dual_problem::Dualization.DualProblem)
+        new(dual_problem, nothing, nothing)
+    end
+    function DualOptimizer(dual_optimizer::MOI.AbstractOptimizer)
+        new(nothing, dual_optimizer, nothing)
+    end
+    function DualOptimizer()
+        new(nothing, nothing, nothing)
+    end
+end
+
+function MOI.supports(::DualOptimizer,
+                      ::Union{MOI.ObjectiveSense,
+                              MOI.ObjectiveFunction{MOI.SingleVariable},
+                              MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}})
+    return true
+end
+
+function MOI.supports_constraint(optimizer::DualOptimizer, F::Type{<:SF}, S::Type{<:SS})
+    if optimizer.dual_optimizer !== nothing
+        MOI.supports_constraint(optimizer.dual_optimizer, F, S)
+    else
+        return true
```

Comment:
This is weird, maybe you should not allow the `dual_optimizer` and `dual_problem` to be `nothing`

Reply:
so, for now the dual model is being built to `dual_problem` and its only passes to `dual_optimizer` when `optimize!` is called.
One possible next step is get rid of the `dual_optimizer` completely and plug the optimizer inside the `dual_problem`, in the `dual_model` field.